src/iiif/ingress_zip_consumer.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+import os
+import shutil
+
+from django.conf import settings
+from ingress.consumer.base import BaseConsumer
+
+from iiif import tools
+from main import settings
+
+logger = logging.getLogger(__name__)
+
+
+class ZipConsumer(BaseConsumer):
+    collection_name = settings.ZIP_COLLECTION_NAME
+
+    """
+    Whether or not to immediately remove messages once consumption succeeds.
+    If set to False, message.consume_succeeded_at will be set.
+    """
+    remove_message_on_consumed = False
+
+    """
+    Whether or not to set Message.consume_started_at immediately once consumption starts
+    """
+    set_consume_started_at = True
+
+    def consume_raw_data(self, raw_data):
+        record = json.loads(raw_data)
+
+        # Create a tmp folder to store downloaded source files
+        zipjob_uuid, tmp_folder_path = tools.create_tmp_folder()
+
+        # Init contents of txt info file which is send along in the zip
+        info_txt_contents = "The following files were requested:\n"
+        for i, (iiif_url, info) in enumerate(record['urls'].items()):
+            info_txt_contents += f"{iiif_url}: "
+            try:
+                file_response, file_url = tools.get_file(
+                    record['request_meta'], info['url_info'], iiif_url, info['metadata'])
+                tools.handle_file_response_errors(file_response, file_url)
+            except tools.ImmediateHttpResponse as e:
+                # Write small file
+                info_txt_contents += f"Not included in this zip: {e.response.content}\n"
+                continue
+            except Exception as e:
+                info_txt_contents += f"Not included in this zip: {e}\n"
+                continue
+
+            # Save image file to tmp folder
+            tools.save_file_to_folder(tmp_folder_path, str(i), file_response.content)
```

Comment:
Files inside the zip are named 1, 2, 3, ... N. They seem to have no extension either. Is that intended, because it seems very user unfriendly.

Reply:
I understand we don't want to expose internal file names, but at least we should add the proper extension. Maybe there is a better solution we can think of than 1, 2, 3, ... N.

Reply:
Je hebt gelijk. Dit was één van de scherpe randjes die nog opgelost moest worden. :-)